src/v/cluster/archival_metadata_stm.h

```diff
@@ -103,7 +100,7 @@ class archival_metadata_stm final : public persisted_stm {
 
     mutex _lock;
 
-    cloud_storage::partition_manifest _manifest;
+    ss::shared_ptr<cloud_storage::partition_manifest> _manifest;
```

Comment:
why does it have to be changed to `shared_ptr`?

Reply:
I removed `#include "cloud_storage/partition_manifest.h"` and added `#include "cloud_storage/fwd.h"` which has only a declaration `class partition_manifest;`. The declaration of `partition_manifest` is incomplete. Object can't be instantiated without the full type. But reference or pointer can. The `ss::shared_ptr<T>` doesn't require `T` to be a complete type. It's possible to use a forward declaration.

Reply:
i don't know if it is worth it, but i'm pretty sure you can use ss::lw_shared_ptr by adding a customization point `lw_shared_ptr_deleter` (see `seastar/core/shared_ptr_incomplete.hh`) it looks like there is an example usage in scylla too for reference.